### PR TITLE
[AAE-20769] Fix precision for incoming values in bigdecimal field

### PR DESCRIPTION
**Pull request by BSekula**
**Please check if the PR fulfills these requirements**

> - [ ] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [x] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

**What is the new behaviour?**

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: lib/core/src/lib/form/components/middlewares/decimal-middleware.service.spec.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        const form = new FormModel();
+        formFieldModel = new FormFieldModel(form, {
+            type: FormFieldTypes.DECIMAL,
+            id: 'id',
+            precision: 3,
+            value: '10.1060'
+        });
+    });
+
+    it('should return field with proepr precisison', () => {
```
Comment (DudaRobert) [inline lib/core/src/lib/form/components/middlewares/decimal-middleware.service.spec.ts:37]:
```suggestion
    it('should return field with proper precisison', () => {
```

File: lib/core/src/lib/form/components/middlewares/decimal-middleware.service.spec.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    });
+
+    it('should round up number, when removed fraction part starts with number 5 ', () => {
+        formFieldModel.value = '10.1035000';
+        formFieldModel.precision = 3;
+        const parsedField = decimalMiddlewareService.getParsedField(formFieldModel);
+        expect(parsedField.value).toBe('10.104');
+    });
+
+    it('should NOT round up number', () => {
```
Comment (DudaRobert) [inline lib/core/src/lib/form/components/middlewares/decimal-middleware.service.spec.ts:58]:
```suggestion
    it('should NOT round up number, when removed fraction part starts with number smaller than 5 ', () => {
```

File: lib/core/src/lib/form/components/middlewares/decimal-middleware.service.spec.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    });
+
+    it('should round up number with correct precisison', () => {
+        formFieldModel.value = '10.1039999';
+        formFieldModel.precision = 3;
+        const parsedField = decimalMiddlewareService.getParsedField(formFieldModel);
+        expect(parsedField.value).toBe('10.104');
+    });
+
+    it('should round up number, when removed fraction part starts with number 5 ', () => {
```
Comment (DudaRobert) [inline lib/core/src/lib/form/components/middlewares/decimal-middleware.service.spec.ts:51]:
```suggestion
    it('should round up number, when removed fraction part starts with number larger than 4 ', () => {
```

**Review by DudaRobert — APPROVED** [review]
Small typo and minor suggestions :)

File: lib/core/src/lib/form/components/middlewares/decimal-middleware.service.spec.ts (hunk 3 above)
Comment (BSekula) [reply lib/core/src/lib/form/components/middlewares/decimal-middleware.service.spec.ts:51]: changed to `when removed fraction part starts with number larger or equal 5`

Want to emphasis the boundary, which is 5 here :)

**Pull request merged**